crates/storage/src/snapshot.rs

```diff
@@ -69,73 +77,103 @@ impl Snapshot {
         &self,
         key: Vec<u8>,
     ) -> Result<(Option<Vec<u8>>, ics23::CommitmentProof)> {
-        let span = Span::current();
-        let snapshot = self.clone();
-
+        let span = tracing::Span::current();
+
+        let (_, substore_config) = self.0.multistore.route_key_bytes(&key);
+        let substore = store::substore::SubstoreSnapshot {
+            config: substore_config,
+            rocksdb_snapshot: self.0.snapshot.clone(),
+            version: self.version(),
```

Comment:
note: we are going to keep separate versioning per substore/tree, the strategy should be something along the lines of keeping track of (prefix -> version) after writes, possibly persisting them on disk.